### DISCUSSION TITLE
Linux Day - Milan - Italy

### DISCUSSION THREAD

**Discussion by deneb-alpha** [uyuni-project/uyuni#7533] [Show and tell]
# info
* https://linuxdaymilano.org/
* CfP: Deadline September  21st  2023

# requirements for the CfP
* Title
* Abstract: max. 1000 chars
* Level (beginner, intermediate, expert)
* Name Surname presenter
* Company (optional)
* email
* telegram contact (optional) 

# attendees
* @deneb-alpha, @mackdk @mbussolotto  (add yourself :D )

# Slides
Uyuni: from server to "macroservice" [Download slides](https://drive.proton.me/urls/1RRXMZFMXC#TnDKpiXi4YUS)

**Comment by mbussolotto** [comment]
## Title
Uyuni: from server to "macroservice"
## Abstract
Uyuni is an  open-source configuration and infrastructure management solution for software-defined infrastructure. In order to run uyuni as a multi-container solution running on Kubernetes, a first fat-container (podman) will be released.
The topic will be:
- share experience about how to containerize an application, 
- how to migrate a deployed application to its containerized version
## Level
Intermediate (some basic knowledge on the topic is needed)
## Name Surname presenter
Michele Bussolotto
## Company (optional)
SUSE

## email
## telegram contact (optional) 

@deneb-alpha @mackdk @cbosdo @rjmateus @admd any suggestion is welcome :)

**Reply by deneb-alpha** [reply]
@mbussolotto in the description I added some details about the needed info for filing the talk proposal. For the abstract, what you wrote is already enough, but if you want you can also expand the text a bit (max 1000 chars)

**Reply by cbosdo** [reply]
Moving Uyuni to microservices may not be desirable in the original meaning of that design concept. I would say the ultimate goal here is to move Uyuni to a multi-container solution running on Kubernetes.

**Reply by mbussolotto** [reply]
> Moving Uyuni to microservices may not be desirable in the original meaning of that design concept. I would say the ultimate goal here is to move Uyuni to a multi-container solution running on Kubernetes.

Fixed, thanks!

**Reply by rjmateus** [reply]
I think you could improve a bit the abstract, and say something like "How can we bring a 10+ codebase to containers and benefit from the most recent infrastructure technologies"  and then give a small inside of the process, which we will start with just one fat container as a step to slip components into other containers.

**Comment by deneb-alpha** [comment]
## Title
Uyuni: the open-source configuration and infrastructure management solution for software-defined infrastructure.
## Abstract
Uyuni is a configuration & infrastructure management tool that saves time, costs and headaches when managing and updating tens, hundreds or even thousands of machines.
With automated patch and package management, it enables the deployment of patches and packages based on software channels and repositories that can be assigned.
Uyuni makes it easier to onboard and manage any Linux server connected to the network, from IoT edge devices to Kubernetes environment, no matter where it is located (private or 3rd party data center or in the public cloud).
Uyuni is a single tool for automated deployment of hardened OS templates (bare metal/VM/container) to tens of thousands of servers and IoT devices for faster, consistent and repeatable provisioning and configuration without compromising speed or security.
The CVE audit feature allows to check the status of public security patches and with OpenSCAP it's also possible to check for specification compliance and apply remediations right from Uyuni.
## Level
Intermediate (some basic knowledge on the topic is needed and I don't think I can classify the proposal as beginner level)
## Name Surname presenter
Marina Latini

## Company (optional)
SUSE

## email
## telegram contact (optional)

**Reply by deneb-alpha** [reply]
@mbussolotto @mackdk @raulillo82 @juliogonzalez @admd @cbosdo what's your take?

**Reply by juliogonzalez** [reply]
> With automated patch and package management, it enables the deployment of patches and packages based on software channels and repositories that can be assigned, helping to reduce costs and time.

Maybe you can mention CVE audit as as well in this context?

About the level, "on the topic" is maybe a bit too generic. Maybe `intermediate knowledge in systems administration`.

Other than that, looks great.

Remember to create a PR to update the news at the site, and email/tweet :)

**Reply by deneb-alpha** [reply]
> Maybe you can mention CVE audit as as well in this context?

actually the CVE audit is also missing from the website. :thinking: 

> About the level, "on the topic" is maybe a bit too generic. Maybe intermediate knowledge in systems administration.

the form has a radiobutton where I can only select the levels, there's no text to be added.

> Remember to create a PR to update the news at the site, and email/tweet :)

yep, as soon as the CfP will close I'll also update website etc with the info.

**Reply by deneb-alpha** [reply]
@juliogonzalez I updated the abstract with the CVE audit and OpenSCAP part but I had to also rephrase the text due to the 1000 chars limit. 
what do you think now? anything else to change?

**Reply by juliogonzalez** [reply]
LGTM!

**Reply by raulillo82** [reply]
LGTM as well!
Small typo correction: %s/Uyuni make it easier/Uyuni makes it easier/g

**Reply by admd** [reply]
LGTM

>  OpenSCAP is it also possible to check for specification compliance and remediation.

...apply the remediations right from Uyuni.

**Reply by admd** [reply]
I am not sure if Demo is involved as you are covering way too much. If it's just presentation then all good.

**Reply by deneb-alpha** [reply]
@admd no demo involved. I had to compress a bit the text because I was hitting the 1000 chars max limit. 

Thanks everyone for the feedback.

**Comment by mackdk** [comment]
## Title

Uyuni contribution made easy

## Abstract

Uyuni is the open-source configuration and infrastructure management solution for software-defined infrastructure. Due to its open-source nature, it welcomes any contribution from its community of enthusiasts. Nevertheless, due to scope and the size of the project it might be very intimidating at first.

This presentation aims to introduce the project and its code base, and give initial directions to a newcomer. In particular, the following topics will be discussed:

- how the community and the project are organised
- what can be contributed
- the structure of the code base and the variety of programming languages used
- the environment set-up for developing and testing
- the submission of a pull request and the code review

## Level

Intermediate

## Name Surname presenter

Thomas Florio

## Company (optional)

SUSE

## E-mail
## Telegram contact (optional)

**Reply by mackdk** [reply]
@deneb-alpha @mbussolotto @juliogonzalez @admd @rjmateus @cbosdo Please feel free to add to the topics and improve

**Reply by deneb-alpha** [reply]
@mackdk I already like it :)

**Reply by cbosdo** [reply]
s/external contributor/newcomer/

**Reply by cbosdo** [reply]
in the pull request and review process you surely mean to mention the CIs and tests we run on the PRs, right? The review process seems like a complex stuff, while you probably want to tell how to understand the failures in the tests.

**Reply by mackdk** [reply]
> in the pull request and review process you surely mean to mention the CIs and tests we run on the PRs, right? The review process seems like a complex stuff, while you probably want to tell how to understand the failures in the tests.

Yes, let me reword it. I meant indeed from the point of view of the submitter, so understanding the checks we run, how to read the results and fix the failures.

**Reply by juliogonzalez** [reply]
I know this is not the main focus of the talk, but you maybe want to explain at some point, that people can contribute even without being developers. 

At the last SUSECon I had a quick presentation telling how to fix bugs at the doc, without even knowing about it. But of course there's also the website, and the weblate translations, bug reporting, helping other uses, etc :-)

**Reply by mackdk** [reply]
> I know this is not the main focus of the talk, but you maybe want to explain at some point, that people can contribute even without being developers.

Yes, I actually plan to mention that as well.

**Comment by deneb-alpha** [comment]
The conference took place https://linuxdaymilano.org/schedule/

We got two slots:
* Uyuni: the open-source configuration and infrastructure management solution for software-defined infrastructure
* Come trasformare un'applicazione server in un'applicazione "macroservizio": l'esempio di Uyuni

we also had an info booth with some gadgets for attendees.
